lib/vmdb/settings_walker.rb

```diff
@@ -50,6 +50,10 @@ def mask_passwords!(settings)
         walk_passwords(settings) { |k, _v, h| h[k] = "********" }
       end
 
+      def filter_passwords!(settings)
+        walk_passwords(settings) { |k, _v, h| h.delete_field(key) }
```

Comment:
This should be `h.delete_field(k)` (as opposed to `key`)

Reply:
Yeah there are bigger issues (OpenStruct/Hash) but will get to this